brian2tools/baseexport/device.py

```diff
@@ -0,0 +1,417 @@
+            if i is not None:
+                connect.update({'i': i})
+            if j is not None:
+                connect.update({'j': j})
+        connect.update({'probability': p, 'n_connections': n,
+                        'synapses': synapses.name,
+                        'source': synapses.source.name,
+                        'target': synapses.target.name, 'type': 'connect'
+                        })
+        # get resolved and clean identifiers
+        strings_with_identifers.append(str(p))
+        strings_with_identifers.append(str(n))
+        identifers_set = set()
+        for string_expr in strings_with_identifers:
+            identifers_set = identifers_set | get_identifiers(string_expr)
+        ident_dict = synapses.resolve_all(identifers_set, namespace)
+        ident_dict = _prepare_identifiers(ident_dict)
+        if ident_dict:
+            connect.update({'identifiers': ident_dict})
+        self.initializers_connectors.append(connect)
+        # update `_connect_called` to allow initialization on
+        # synaptic variables
+        synapses._connect_called = True
+
+    def build(self, direct_call=True, debug=False):
+        """
+        Collect all run information and export the standard
+        dictionary format
+
+        Parameters
+        ----------
+        direct_call: bool, optional
+            To check the call to build() was made directly
+
+        debug: bool, optional
+            To build the device in debug mode
+        """
+        # buil_on_run = True but called build() directly
+        if self.build_on_run and direct_call:
+            raise RuntimeError('You used set_device with build_on_run=True '
+                               '(the default option), which will '
+                               'automatically build the simulation at the '
+                               'first encountered run call - do not call '
+                               'device.build manually in this case. If you '
+                               'want to call it manually, '
+                               'e.g. because you have multiple run calls, use'
+                               ' set_device with build_on_run=False.')
+        # if already built
+        if self.has_been_run:
+            raise RuntimeError('The network has already been built and run '
+                               'before. To build several simulations in '
+                               'the same script, call "device.reinit()" '
+                               'and "device.activate()". Note that you '
+                               'will have to set build options (e.g. the '
+                               'directory) and defaultclock.dt again.')
+        # change the flag
+        self.has_been_run = True
+        # if to operate in debug mode
+        if debug:
+            logger.debug("Building ExportDevice in debug mode")
+            # print dictionary format using pprint
+            np.set_printoptions(threshold=10)
```

Comment:
Small thing, but it would be good to reset the print options afterwards, otherwise they are changed for the remainder of the Python session.